### ISSUE TITLE
Swapping position of two drivers loses position of last one

### ISSUE THREAD

**Issue by EricAndrechek** [f1-tools/MVF1-Streamdeck#7]
When two driver feeds are being swapped, only one of them successfully swaps. This is due to the way the underlying swap logic is built and needs to be fixed.

**Comment by ManCaveMedia** [comment]
Can't reproduce this on the StreamDeck XL on v1.1.0.
Has this been fixed yet?

**Comment by jonahhb** [comment]
I just double-checked and I can recreate _part of it_. I think the thing is if you only have two players open and switch them.

This causes the eventual sync (in waitToSync) to fail because the 'oldest player' no longer exists so there is nothing to sync to.

**Comment by ManCaveMedia** [comment]
Could this be solved by temporarily storing the details for the oldest stream and use that to sync the swapped players to if there's only 2 or less streams?

**Comment by jonahhb** [comment]
I don't think so because the sync streams endpoint on the MV API takes a player ID and then syncs all the other players to that player. 

Unfortunately, we cannot give the endpoint a timestamp (or other 'storable' information), but only a player ID.

**Comment by EricAndrechek** [comment]
So to clarify @jonahhb, does this issue arise when there is no longer an existing oldest player? Can we build an error handling bit to use the next "oldest"?

**Comment by jonahhb** [comment]
@EricAndrechek It happens when there are no other players (other than the ones being switched) so there is neither the oldest player nor the next oldest player to sync to. There are just no players at all alive with the place to sync too. 

Potentially we could not delete the old players until the wait to sync is done but then I worry about people with computers that cannot handle many players and having the extra player open might make things laggy.

**Comment by EricAndrechek** [comment]
I see, thanks. How does Multiviewer handle this case if you have things playing, close everything, and open a new player? You have to sync to a specific window still, right? So it wouldn't know where you were either?
